File: backend/src/main/java/com/bv87/diemdanh/dto/ScanLogPageDto.java
package com.bv87.diemdanh.dto;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

/** Paginated scan logs for one employee on one calendar day (VN). */
@Getter
@Builder
public class ScanLogPageDto {
    private final Integer empCode;
    private final String empCodeFormatted;
    private final String fullname;
    private final LocalDate date;
    private final List<ScanLogItemDto> items;
    private final int page;
    private final int pageSize;
    private final long totalItems;
    private final int totalPages;
}

File: desktop/src/BV87.Core/Models/Attendance/ScanLogPageDto.cs
namespace BV87.Core.Models.Attendance;

/// <summary>Paginated scan logs for one employee on one calendar day.</summary>
public sealed class ScanLogPageDto
{
    public int? EmpCode { get; set; }
    public string? EmpCodeFormatted { get; set; }
    public string? Fullname { get; set; }
    public DateOnly? Date { get; set; }
    public List<ScanLogItemDto> Items { get; set; } = [];
    public int Page { get; set; }
    public int PageSize { get; set; }
    public long TotalItems { get; set; }
    public int TotalPages { get; set; }
}
